Data/HashMap/Base.hs

```diff
@@ -1883,6 +1890,33 @@ fromListWith :: (Eq k, Hashable k) => (v -> v -> v) -> [(k, v)] -> HashMap k v
 fromListWith f = L.foldl' (\ m (k, v) -> unsafeInsertWith f k v m) empty
 {-# INLINE fromListWith #-}
 
+-- | /O(n*log n)/ Construct a map from a list of elements.  Uses
+-- the provided function to merge duplicate entries.
+--
+-- === Examples
+--
+-- Given a list of key-value pairs where the keys are of different flavours, e.g:
+--
+-- > data Key = Mul | Add
+--
+-- and the values need to be combined differently on a collision, depending on the key:
```

Comment:
`collision` sounds like hash collision, but I don't think that's what you mean.

Reply:
Good point. I'll update the docs.